plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/generator/SQLGeneratorDialog.java

```diff
@@ -278,16 +281,21 @@ public void widgetSelected(SelectionEvent e) {
                 public void widgetSelected(SelectionEvent e) {
                     sqlGenerator.setShowCastParams(supportsCastParamsButton.getSelection());
                     getDialogBoundsSettings().put(DBPScriptObject.OPTION_CAST_PARAMS, supportsCastParamsButton.getSelection());
-
-                    UIUtils.runInUI(sqlGenerator);
-                    Object sql = sqlGenerator.getResult();
-                    if (sql != null) {
-                        setSQLText(CommonUtils.toString(sql));
-                        updateSQL();
-                    }
+                    startGenerateJob();
                 }
             });
         }
         return composite;
     }
+
+    private void startGenerateJob() {
```

Comment:
I'd prefer to call it like this
```suggestion
    private void startGenerateDDLJob() {
```